test/teiserver/game/balancer_server_test.exs

```diff
@@ -35,8 +40,176 @@ defmodule Teiserver.Game.BalancerServerTest do
+      |> Enum.map(fn {_user, index} ->
+        %{
+          "name" => "user" <> to_string(index + 1) <> "_" <> to_string(ExULID.ULID.generate()),
+          "email" => to_string(ExULID.ULID.generate()) <> "@example.com",
+          "permissions" => []
+        }
+      end)
+      |> Enum.map(fn x -> make_user(x) end)
+      # unwrap user creation response
+      |> Enum.map(fn {:ok, reply} -> reply end)
+      |> Enum.map(fn x -> Map.put(x, :userid, x.id) end)
+
+    # set the user rank
+    for {user, rank} <- Enum.zip(users, list_of_ranks) do
+      Teiserver.Account.update_user_stat(user.id, %{
+        :rank => rank
+      })
+    end
+
+    # set the user party_id
+    users =
+      for {user, party_id} <- Enum.zip(users, list_of_parties) do
+        Map.put(user, :party_id, party_id)
+      end
+
+    users
+  end
+
+  test "load some players in and run a balance pass or two" do
+    team_count = 2
+
+    players =
+      make_users_with_ranks_and_parties([10, 20, 30, 40, 50, 60], [1, 2, 3, 4, 5, 6])
+      |> Enum.map(fn x -> Map.put(x, :userid, x.id) end)
+
+    {:ok, pid} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    first_balance_pass =
+      GenServer.call(
+        pid,
+        {:make_balance, team_count, [], players}
+      )
+
+    second_balance_pass_hash_reuse =
+      GenServer.call(
+        pid,
+        {:make_balance, team_count, [], players}
+      )
+
+    assert second_balance_pass_hash_reuse.hash == first_balance_pass.hash
+
+    player_list_with_one_less_player = tl(players)
+
+    third_balance_pass_with_fewer_players =
+      GenServer.call(
+        pid,
+        {:make_balance, team_count, [], player_list_with_one_less_player}
+      )
+
+    assert second_balance_pass_hash_reuse.hash != third_balance_pass_with_fewer_players.hash
+  end
+
+  test "get_current_balance returns a result" do
+    team_count = 2
+
+    players = make_users_with_ranks_and_parties([10, 20, 30, 40], [1, 2, 3, 4])
+
+    {:ok, pid} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    GenServer.call(
+      pid,
+      {:make_balance, team_count, [], players}
+    )
+
+    refute GenServer.call(pid, :get_current_balance) == nil
+  end
+
+  test "reset_hashes clears hash and result" do
+    team_count = 2
+
+    players = make_users_with_ranks_and_parties([10, 20, 30, 40], [1, 2, 3, 4])
+
+    {:ok, pid} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    GenServer.call(
+      pid,
+      {:make_balance, team_count, [], players}
+    )
+
+    refute GenServer.call(pid, :get_current_balance) == nil
+
+    with {:ok, value} <- Map.fetch(GenServer.call(pid, :report_state), :last_balance_result) do
+      refute value == nil
+    end
+
+    GenServer.cast(
+      pid,
+      :reset_hashes
+    )
+
+    poll_until_nil(fn -> GenServer.call(pid, :get_current_balance) end)
+
+    with {:ok, hash} <- Map.fetch(GenServer.call(pid, :report_state), :last_balance_hash) do
+      assert hash == nil
+    end
```

Comment:
you can do `assert Map.get(GenServer.call(pid, :report_state), :last_balance_hash, :not_found) == nil`

Reply:
Ah, yes, that is cleaner, thank you.